subsys/bluetooth/mesh/rpr_cli.c

```diff
@@ -358,7 +365,7 @@ static void link_timeout(struct k_work *work)
 	struct bt_mesh_rpr_cli *cli = CONTAINER_OF(k_work_delayable_from_work(work),
 						   struct bt_mesh_rpr_cli, link.timeout);
 
-	if (cli->link.state != BT_MESH_RPR_LINK_IDLE) {
+	if (bearer.link != BEARER_LINK_IDLE) {
```

Comment:
Why callback about closing link is based on bearer state rather than link state? Will server return correct RPState in Link Status on Link Get here?

Reply:
Because `cli->link.state` only changes by a message received from the server. If nothing was received from the server, the state will be `BT_MESH_RPR_LINK_IDLE` and by the timeout the client won't do anything. This is explained in the corresponding commit message.

I see I could use `LINK_ACTIVE` flag from `bt_mesh_prov_link.flags` like `pb_gatt.c` does, but:
1. it will add dependency on `bt_mesh_prov_link` in `rpr_cli.c`. Though maybe it is fine.
2. I still need to keep the bearer link state separately. This is needed to avoid double call to `bearer.cb->link_opened` and `link_closed` callbacks if another message with the same link state arrives from the server (this is not an issue for pb-gatt and pb-adv). Adding something like `LINK_OPENING` flag to `bt_mesh_prov_link` looks a bit weird because this one and `LINK_ACTIVE` are mutually exclusive and I don't think flags should be used in this case.